web/packages/teleport/src/services/api/api.ts

```diff
@@ -77,6 +77,15 @@ const api = {
     );
   },
 
+  deleteWithHeaders(url, headers?: Record<string, string>, signal?) {
+    return api.fetch(url, {
+      method: 'DELETE',
+      headers,
+      signal,
+    });
+  },
+
+  // TODO (avatus) add abort signal to this
```

Comment:
added this TODO because adding the abort signal to our `put` method here would change a LOT of spots and I don't want it to muddy up this PR

Reply:
You can make it optional